FWCore/Concurrency/src/SerialTaskQueue.cc

```diff
@@ -23,98 +23,93 @@ using namespace edm;
 //
 // member functions
 //
-SerialTaskQueue::~SerialTaskQueue()
-{
-  //be certain all tasks have completed
+SerialTaskQueue::~SerialTaskQueue() {
+  // be certain all tasks have completed
   bool isEmpty = m_tasks.empty();
   bool isTaskChosen = m_taskChosen;
-  if ( (not isEmpty and not isPaused()) or isTaskChosen) {
-    pushAndWait([]() {return;});
+  if ((not isEmpty and not isPaused()) or isTaskChosen) {
+    pushAndWait([]() { return; });
   }
 }
 
-bool
-SerialTaskQueue::resume() {
-  if(0==--m_pauseCount) {
+bool SerialTaskQueue::resume() {
+  if (0 == --m_pauseCount) {
     tbb::task* t = pickNextTask();
-    if(nullptr != t) {
+    if (nullptr != t) {
       tbb::task::spawn(*t);
     }
     return true;
   }
   return false;
 }
 
-void
-SerialTaskQueue::pushTask(TaskBase* iTask) {
+void SerialTaskQueue::pushTask(TaskBase* iTask) {
   tbb::task* t = pushAndGetNextTask(iTask);
-  if(nullptr!=t) {
-    tbb::task::spawn(*t);      
+  if (nullptr != t) {
+    tbb::task::spawn(*t);
   }
 }
 
-tbb::task* 
-SerialTaskQueue::pushAndGetNextTask(TaskBase* iTask) {
+tbb::task* SerialTaskQueue::pushAndGetNextTask(TaskBase* iTask) {
   tbb::task* returnValue{nullptr};
-  if likely(nullptr!=iTask) {
-    m_tasks.push(iTask);
-    returnValue = pickNextTask();
-  }
+  if
+    likely(nullptr != iTask) {
+      m_tasks.push(iTask);
+      returnValue = pickNextTask();
+    }
   return returnValue;
 }
 
-
-tbb::task*
-SerialTaskQueue::finishedTask() {
+tbb::task* SerialTaskQueue::finishedTask() {
   m_taskChosen.store(false);
   return pickNextTask();
 }
 
-SerialTaskQueue::TaskBase*
-SerialTaskQueue::pickNextTask() {
-  
+SerialTaskQueue::TaskBase* SerialTaskQueue::pickNextTask() {
   bool expect = false;
-  if likely(0 == m_pauseCount and m_taskChosen.compare_exchange_strong(expect,true)) {
-    TaskBase* t=nullptr;
-    if likely(m_tasks.try_pop(t)) {
-      return t;
-    }
-    //no task was actually pulled
-    m_taskChosen.store(false);
-    
-    //was a new entry added after we called 'try_pop' but before we did the clear?
-    expect = false;
-    if(not m_tasks.empty() and m_taskChosen.compare_exchange_strong(expect,true)) {
-      TaskBase* t=nullptr;
-      if(m_tasks.try_pop(t)) {
-        return t;
-      }
-      //no task was still pulled since a different thread beat us to it
+  if
+    likely(0 == m_pauseCount and m_taskChosen.compare_exchange_strong(expect, true)) {
```

Comment:
this looks worse than the original, IMHO.
An extra set of parentheses ```if (likely(expr) ) {``` would look more sensible to me